File: benchmark/c/cpw/minimum_perimeter_n_blocks/minimum_perimeter_n_blocks.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int minimum_perimeter_n_blocks ( int n ) ;
int minimum_perimeter_n_blocks ( int n ) {
  int l = sqrt ( n );
  int sq = l * l;
  if ( sq == n ) return l * 4;
  else {
    long int row = n / l;
    long int perimeter = 2 * ( l + row );
    if ( n % l != 0 ) perimeter += 2;
    return perimeter;
  }
}

File: translation_result/else_bms/fail/c/cpw/minimum_perimeter_n_blocks/gemini/Hinted/minimum_perimeter_n_blocks.rs

use std::f64;

fn minimum_perimeter_n_blocks(n: i32) -> i32 {
    let l = (n as f64).sqrt() as i32;
    let sq = l * l;
    if sq == n {
        return l * 4;
    } else {
        let row = n / l as i32;
        let mut perimeter = 2 * (l + row);
        if n % l != 0 {
            perimeter += 2;
        }
        return perimeter;
    }
}
